### DISCUSSION TITLE
Patterns

### DISCUSSION THREAD

**Discussion by ryedunn** [marlonrichert/zsh-hist#15] [Q&A] [answered]
I stumbled upon this from a post on SO and this is exactly what I was looking for.  Unfortunately, Im a bit confused with some of the functionality.  
Patterns for example, if I wanted to remove all 'ls' commands...  I thought I could use something like `hist d ^ls\s`, but this doesnt seem to work as I would have hoped.  Although I can confirm your example in the documentation `hist d $'*(\n|;)'` works just fine. 
Any suggestions on what Im doing wrong here?

**Comment by marlonrichert** [comment, marked as answer]
* The pattern syntax is not regex, but [glob expressions](https://zsh.sourceforge.io/Doc/Release/Expansion.html#Glob-Operators). This is the same syntax that's used for [the `$HISTORY_IGNORE` parameter](https://zsh.sourceforge.io/Doc/Release/Parameters.html#index-HISTORY_005fIGNORE).
* You need to quote the pattern or else `zsh` will perform [filename generation](https://zsh.sourceforge.io/Doc/Release/Expansion.html#Filename-Generation) on your argument before passing it to `hist`.

So, for your case, to remove all lines starting with the word `ls`, you should use `hist d 'ls *'`.
